tests/test_rest_api.py

```diff
@@ -199,6 +204,18 @@ def test_custom_dataset_plugin(airtemp_ds, dataset_plugin):
     assert list(json_response['variables'].keys()) == list(airtemp_ds.variables.keys())
 
 
+def test_uninitialized_plugin(uninitialized_dataset_plugin):
+    """Checks for custom AttributeError message when plugin is not initialized."""
+    rest = Rest({})
+
+    try:
+        rest.register_plugin(uninitialized_dataset_plugin)
+    except AttributeError as e:
+        assert 'Plugin' in str(e)
+    else:
+        assert False, 'Expected AttributeError'
+
+
```

Comment:
Thanks for adding a test for uninitialized plugins! 

This may be easier to understand with [`pytest.raises`](https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions).

```suggestion
def test_uninitialized_plugin(uninitialized_dataset_plugin):
    """Checks for custom AttributeError message when plugin is not initialized."""
    rest = Rest({})

    with pytest.raises(AttributeError) as excinfo:
        rest.register_plugin(uninitialized_dataset_plugin)
    assert 'Plugin' in excinfo.value
```

Reply:
Ooh cool didn't realize you could do that!

Reply:
Implemented w/ commit 1cccd01eb924ef44d4fca1ee4ad79d9f4aef227c